### PR TITLE
Added spiner and progress bar, apart from few improvements in code

### LINKED ISSUE #4: feature: add progress bar or spinner
Once processing has started, theere is no way to know what-s happening.

image

### PR DESCRIPTION
**Pull request by omotto**
Added spinner and progress bar:

image

### CODE REVIEW

File: reorganize.py [hunk 1]
```diff
@@ -59,11 +59,13 @@ def init_logger(level):
...
+    logging.debug("[+] Parameters: source= %s, destination= %s, extennsions= %s, debug= %s", source, destination,
+                  extensions, debug)
     image_organizer = FileOrganizer(source, destination, extensions)
     try:
         logging.debug("[+] Starting")
         image_organizer.start()
-    except Exception:
+    except Exception as err:
+        logging.debug(f"Unexpected {err=}, {type(err)=}")
         logging.error("[-] Something went really wrong")
```
Comment (eballo) [inline reorganize.py]: Do we really want two logs in here ? debug and error ? would not be better to update the error and add value if needed ?

File: src/organizer.py [hunk 2]
```diff
@@ -6,43 +6,32 @@
 from src.process import FileProcessor
 import logging
 
-from src.utils import do_you_want_to_continue
+import src.utils
```
Comment (eballo) [inline src/organizer.py:9]: nice one ;)

File: src/organizer.py [hunk 3]
```diff
@@ -6,43 +6,32 @@
...
-
-    @staticmethod
-    def get_list_missing_files(
-        all_files: List[str], processed_files: List[str]
-    ) -> List[str]:
-        return [x for x in all_files if x not in processed_files]
+                processed_files.append(filename.lower())
+        we.run(end=True)
+        missing_files = [x for x in all_files if x not in processed_files]
+        return [all_files, processed_files, missing_files]
```
Comment (eballo) [inline src/organizer.py:34]: +1 this I wanted to do it :D I thought about this improvement! nice one!

**Review by eballo — APPROVED** [review]
Looking really good!

File: reorganize.py (hunk 1 above)
Comment (omotto) [reply reorganize.py]: From my understanding logging,error is going to show the error when it happens, but error.debug when debug flag is enabled only. I do not want to show the `err` in "normal" mode, but when I'm debugging I want to know what happened.

Comment (omotto) [reply reorganize.py]: Changed as:

```
    except Exception as err:
        if debug == logging.DEBUG:
            error_msg = f"Unexpected {err=}, {type(err)=}"
        else:
            error_msg = "[-] Something went really wrong"
        logging.error(error_msg)
```

**Pull request merged**